File: 0202-happy-number/0202-happy-number.py
class Solution:
    def sumSquareDigits(self, n: int) -> int:
        count = 0
        while n:
            value = n%10
            count += value * value
            n = n//10
        return count

    def isHappy(self, n: int) -> bool:
        mySet = set()
        while n != 1:
            n = self.sumSquareDigits(n)
            if n in mySet:
                return False
            mySet.add(n)
        return True

File: 0202-happy-number/0202-happy-number.cpp
class Solution {
public:
    int sumSquareDigits(int num){
        int temp = 0;
        while(num){
            int element = num%10;
            temp += element * element;
            num /= 10;
        }

        return temp;
    }

    bool isHappy(int n) {
        unordered_set<int> mySet;

        while(n != 1){
            n = sumSquareDigits(n);
            cout<<n<<endl;
            if(mySet.count(n) > 0)
                return false;
            mySet.insert(n);
        }

        return true;
    }
};

/*


*/